File: Plugins/Decoders/UnShortenURL.py
from Plugins import Plugin

import requests

class UnShortenURL(Plugin.Plugin):
  def __init__(self, url: str = None, name = "UnShorten URL"):
    super().__init__(name)
    self._url = url.strip() if url is not None else None

  def run(self):
    print("\n --------------------------------- ")
    print("   U R L   U N S H O R T E N E R  ")
    print(" --------------------------------- ")
    if self._url == None:
      self._url = str(input('Enter URL: ').strip())
    req = requests.get(str('https://unshorten.me/s/' + self._url))
    print(req.text)

File: src/plugins/decoders/unshorten_url.rs
use super::Plugin;
use serde_json;
use reqwest;

use crate::utils::get_input;

pub struct UnshortenUrl {}

impl Plugin for UnshortenUrl {
  const NAME: &'static str = "Unshorten URL";

  fn new() -> Self {
    Self {}
  }

  fn perform_lookup(&self, value: &str) -> Result<serde_json::Value, String> {
    let url = format!("https://unshorten.me/s/{}", value);
    let response = reqwest::blocking::get(&url).unwrap();

    if response.status() == reqwest::StatusCode::OK {
      let body = response.text().unwrap();
      Ok(serde_json::json!({"status": "complete", "data": body}))
    } else {
      Err(String::from("query to urlshorten.me failed"))
    }
  }

  fn run(&self) {
    println!("{}", Self::NAME);
  
    let usr_in = get_input("Enter URL to decode: ");
    match self.perform_lookup(&usr_in.trim()) {
      Ok(result) => println!("\n{}", result),
      Err(error) => eprintln!("Error: {}", error)
    }
  }
}